src/lib/getVisibilityWatcher.ts

```diff
@@ -63,11 +63,20 @@ const removeChangeListeners = () => {
 
 export const getVisibilityWatcher = () => {
   if (firstHiddenTime < 0) {
-    // If the document is hidden when this code runs, assume it was hidden
-    // since navigation start. This isn't a perfect heuristic, but it's the
-    // best we can do until an API is available to support querying past
-    // visibilityState.
-    firstHiddenTime = initHiddenTime();
+    // Check if we have a previous hidden `visibility-state` performance entry.
+    /* eslint-disable indent */
+    const firstVisibilityStateHiddenTime = !document.prerendering
+      ? globalThis.performance
+          .getEntriesByType('visibility-state')
+          .filter((e) => e.name === 'hidden')[0]?.startTime
```

Comment:
This works.

Arguably, though, if the browser supports `visibility-state` entries then you don't need the `visibilitychange` listeners, and you can just always get the first hidden time from the first entry in the `firstHiddenTime()` getter, rather than there.

Reply:
That's not quite true in the case of bfcache where you wanna ignore times prior to the restore.

It also allows you support partial prerenders that are activated and moved away from before FCP/LCP.

So good to leave it as is? I could also added a comment if you wanted?